### DISCUSSION TITLE
[Design/General] PDA Messenger Feature Expansion

### DISCUSSION THREAD

**Discussion by Contrabang** [ParadiseSS13/Paradise#21385] [Documents]
Everyone uses radios. That's a guarantee, its the easiest way to communicate in game, with everyone. PDAs are useful sometimes for communicating information directly to one person, but thats about it. So, what if we added some real life texting-like features.
# General Idea
Overhaul of PDA messaging code, because its pretty awful. Make it use special datums for conversations and messages, with actual vars for holding data instead of an awkwardly formatted list.
New feature Ideas:
- Blocking people (They can't message you, but they can see your PDA.)
- Muting people (They can message you, but you won't be notified of any new messages from them)
- Group chats (Being able to add up to a certain amount of people to a groupchat (say 10 or so to stop someone from adding the entire station to a group chat)). Cool if you want to have a departmental group chat, or just something with your friends.
- A special PDA cartridge allowing you to upload photos into your PDA and send them to people (A pda cartridge being useful? Preposterous)
- Special departmental PDAs that are a part of the wall mounted request terminals, allowing you to send either public (will announce over departmental comms) or private (announce that there is a new message, but not what the message is, and is toggleable from the sender) to the request terminal. These PDAs can be turned off (or mute someone) in the console if someone is abusing it by spamming messages. These PDAs would not be able to be added to group chats.
- The AI being able to group chat also, and getting a PDA messenger overhaul, because holy crap the AI's PDA system sucks.

And a few other weird ideas, but these are the main ones I have in mind. I'm not commited to all of them, but they're fun ideas. I know TG has tackled similar things to their with their messenger system, so I'm going to look there to get some inspiration codewise.

**Comment by TaM-tHe-JaM** [comment]
I, personally, find this to be a marvelous idea and would love to see it implemented.
